src/index.cpp

```diff
@@ -4395,6 +4396,28 @@ void Index::compute_sort_scores(const std::vector<sort_by>& sort_fields, const i
             scores[0] = int64_t(found);
         } else if(field_values[0] == &vector_distance_sentinel_value) {
             scores[0] = float_to_int64_t(vector_distance);
+        } else if(field_values[0] == &vector_query_sentinel_value) {
+            scores[0] = float_to_int64_t(2.0f);
+            try {
+                auto& field_vector_index = vector_index.at(sort_fields[0].vector_query.field_name);
+                const auto& values = field_vector_index->vecdex->getDataByLabel<float>(seq_id);
+                const auto& dist_func = field_vector_index->space->get_dist_func();
+                float dist = 2.0f;
+                if(field_vector_index->distance_type == cosine) {
+                    std::vector<float> normalized_values(sort_fields[0].vector_query.values.size());
+                    hnsw_index_t::normalize_vector(sort_fields[0].vector_query.values, normalized_values);
```

Comment:
We should do this normalization in pre-processing. Otherwise we are doing this for every item found.